packages/nx/src/utils/package-manager.ts

```diff
@@ -29,19 +29,26 @@ export interface PackageManagerCommands {
   run: (script: string, args: string) => string;
 }
 
+/**
+ * Detects which package manager is used in the workspace based on the lock file.
+ */
+function getPackageManager(dir: string = '') {
+  return existsSync(join(dir, 'yarn.lock'))
+    ? 'yarn'
+    : existsSync(join(dir, 'bun.lockb'))
+    ? 'bun'
+    : existsSync(join(dir, 'pnpm-lock.yaml')) ||
+      existsSync(join(dir, 'pnpm-workspace.yaml'))
+    ? 'pnpm'
+    : 'npm';
+}
+
 /**
```

Comment:
Added bun lock logic in util function

Reply:
I think the bun lock file check should come first.

According to https://bun.sh/docs/install/lockfile a feature of bun is that it will generate a yarn.lock file for users to more easily inspect. So the presence of a yarn.lock file could actually still mean that it is a bun repo (as long as there is a also a `bun.lockb` present).

Therefore I think the simplest thing is to simply apply the check for `existsSync(join(dir, 'bun.lockb'))` first and return `bun`. Please also add a comment in the code with this same explanation as to why we check for bun first

Reply:
However please see my next comment about returning this logic to its original location